### PR TITLE
fix: goroutine leaks

### PR DESCRIPTION
**Pull request by miparnisari**
Gubernator has some goroutine leaks stemming from not properly closing resources. These leaks are not many. However, any  software that uses gubernator and that wants to assert that it has 0 leaks will get leaks coming from gubernator 😅

### CODE REVIEW

File: cluster/cluster_test.go [hunk 1]
```diff
@@ -23,12 +23,16 @@ import (
 	"github.com/mailgun/gubernator/v2/cluster"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
 )
 
 func TestStartMultipleInstances(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
```
Comment (miparnisari) [inline cluster/cluster_test.go:32]: ideally this check is in every test, but i'm starting small here.

File: go.sum [hunk 2]
```diff
@@ -881,12 +882,11 @@ k8s.io/apimachinery v0.23.3/go.mod h1:BEuFMMBaIbcOqVIJqNZJXGFTP4W6AycEpb5+m/97hr
 k8s.io/client-go v0.23.3 h1:23QYUmCQ/W6hW78xIwm3XqZrrKZM+LWDqW2zfo+szJs=
 k8s.io/client-go v0.23.3/go.mod h1:47oMd+YvAOqZM7pcQ6neJtBiFH7alOyfunYN48VsmwE=
 k8s.io/gengo v0.0.0-20210813121822-485abfe95c7c/go.mod h1:FiNAH4ZV3gBg2Kwh89tzAEV2be7d5xI0vBa/VySYy3E=
-k8s.io/klog v0.3.1 h1:RVgyDHY/kFKtLqh67NvEWIgkMneNoIrdkN0CxDSQc68=
```
Comment (miparnisari) [inline go.sum:884]: this is from 2019! 😱 https://github.com/kubernetes/klog/releases/tag/v0.3.1

Comment (Baliedge) [reply go.sum:884]: Seems unused, anyway.  Everything uses `/v2` variant.
```
$ go mod graph | grep k8s.io/klog
github.com/mailgun/gubernator/v2 k8s.io/klog@v0.3.1
github.com/mailgun/gubernator/v2 k8s.io/klog/v2@v2.30.0
k8s.io/apimachinery@v0.23.3 k8s.io/klog/v2@v2.30.0
k8s.io/client-go@v0.23.3 k8s.io/klog/v2@v2.30.0
k8s.io/klog/v2@v2.30.0 github.com/go-logr/logr@v1.2.0
k8s.io/kube-openapi@v0.0.0-20211115234752-e816edb12b65 k8s.io/klog/v2@v2.2.0
k8s.io/utils@v0.0.0-20211116205334-6203023598ed k8s.io/klog/v2@v2.0.0
k8s.io/gengo@v0.0.0-20210813121822-485abfe95c7c k8s.io/klog/v2@v2.2.0
k8s.io/klog/v2@v2.2.0 github.com/go-logr/logr@v0.2.0
k8s.io/utils@v0.0.0-20210802155522-efc7438f0176 k8s.io/klog/v2@v2.0.0
k8s.io/klog/v2@v2.0.0 github.com/go-logr/logr@v0.1.0
```

File: functional_test.go [hunk 3]
```diff
@@ -64,8 +71,19 @@ func TestMain(m *testing.M) {
...
 	}
-	defer cluster.Stop()
-	os.Exit(m.Run())
+	code := m.Run()
+	cluster.Stop()
+
+	// os.Exit doesn't run deferred functions
+	os.Exit(code)
+
+	//goleak.VerifyTestMain(
```
[the remark this answers is not in the document]
Comment (miparnisari) [reply functional_test.go:80]: No, but if this PR is approved i was planning on sending a follow-up with this code block uncommented.

File: gubernator.go [hunk 4]
```diff
@@ -169,6 +167,10 @@ func (s *V1Instance) Close() (err error) {
 		return errors.Wrap(err, "Error in workerPool.Close")
 	}
 
+	//for _, client := range s.conf.LocalPicker.Peers() {
```
[the remark this answers is not in the document]
Comment (miparnisari) [reply gubernator.go:170]: Not for this PR, for a follow-up.

File: functional_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Baliedge) [reply functional_test.go:80]: You're safe to remove it without breaking approval.

**Comment by Baliedge** [comment]
@miparnisari Not sure if you've noticed.  There's still a failing test that seems relevant to your changes.

**Comment by miparnisari** [comment]
@Baliedge please see https://github.com/mailgun/gubernator/pull/223 for those failures

**Review by thrawn01 — COMMENTED** [review]
This looks awesome! thanks for doing this.

**Comment by Baliedge** [comment]
@miparnisari You'll need to fix a merge conflict, then I'll merge.

**Review by Baliedge — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**